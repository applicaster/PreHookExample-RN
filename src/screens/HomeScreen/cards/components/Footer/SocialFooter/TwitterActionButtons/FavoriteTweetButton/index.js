import { connect } from 'react-redux';

import FavoriteTweetButton from './FavoriteTweetButton';
import { getFavoriteTweets } from '../../../../../../../../selectors';

const mapStateToProps = state => ({
  favorites: getFavoriteTweets(state) || {},
});

export default connect(mapStateToProps, undefined)(FavoriteTweetButton);
