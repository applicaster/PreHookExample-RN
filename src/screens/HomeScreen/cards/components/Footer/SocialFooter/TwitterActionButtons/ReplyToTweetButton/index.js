import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setActiveEventId, toggleModal } from '../../../../../../../../actions';

import ReplyToTweetButton from './ReplyToTweetButton';

const mapDispatchToProps = dispatch => bindActionCreators({
  setActiveEventId,
  toggleModal,
}, dispatch);

export default connect(undefined, mapDispatchToProps)(ReplyToTweetButton);
