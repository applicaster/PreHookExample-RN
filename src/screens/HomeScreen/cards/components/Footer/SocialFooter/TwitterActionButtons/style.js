import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  twitterActionButtons: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});
