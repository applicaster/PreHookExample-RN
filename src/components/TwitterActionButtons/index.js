import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  View,
 } from 'react-native';
import { styles } from './style';
import EventDetailCount from '../EventDetailCount';
import RetweetButton from '../RetweetButton';
import ReplyToTweetButton from '../ReplyToTweetButton';

class TwitterActionButtons extends Component {
  render() {
    const { favoriteCount, retweetCount, eventId } = this.props;
    return (
      <View style={[styles.twitterActionButtons]}>
        <ReplyToTweetButton key={'replyButton'} eventId={eventId} />
        <RetweetButton key={'retweetButton'} eventId={eventId} retweetCount={retweetCount} />
        <EventDetailCount key={'favorites'} eventId={eventId} count={favoriteCount} label={'Favorites'} />
      </View>
    );
  }
}

TwitterActionButtons.propTypes = {
  eventId: PropTypes.string,
  favoriteCount: PropTypes.number,
  retweetCount: PropTypes.number,
};

export default TwitterActionButtons;
