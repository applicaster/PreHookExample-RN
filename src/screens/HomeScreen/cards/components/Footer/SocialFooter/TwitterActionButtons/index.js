import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  View,
 } from 'react-native';
import { styles } from './style';
import RetweetButton from './RetweetButton';
import ReplyToTweetButton from './ReplyToTweetButton';
import FavoriteTweetButton from './FavoriteTweetButton';

class TwitterActionButtons extends Component {
  render() {
    const { favoriteCount, retweetCount, eventId } = this.props;
    return (
      <View style={[styles.twitterActionButtons]}>
        <ReplyToTweetButton key={'replyButton'} eventId={eventId} />
        <RetweetButton key={'retweetButton'} eventId={eventId} retweetCount={retweetCount} />
        <FavoriteTweetButton key={'favoriteTweetButton'} eventId={eventId} favoriteCount={favoriteCount} />
      </View>
    );
  }
}

TwitterActionButtons.propTypes = {
  eventId: PropTypes.string.isRequired,
  favoriteCount: PropTypes.number.isRequired,
  retweetCount: PropTypes.number.isRequired,
};

export default TwitterActionButtons;
