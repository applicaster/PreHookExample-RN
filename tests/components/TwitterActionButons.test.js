import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TwitterActionButtons from '../../src/components/TwitterActionButtons';

jest.mock('../../src/components/EventDetailCount', () => 'EventDetailCount');
jest.mock('../../src/components/ReplyToTweetButton', () => 'ReplyToTweetButton');
jest.mock('@applicaster/feed-rn-utils', () => 'FeedRNUtils');

describe('TwitterActionButtons', () => {
  test('component renders correctly', () => {
    const html = renderer.create(
      <TwitterActionButtons />
    ).toJSON();
    expect(html).toMatchSnapshot();
  });
});
