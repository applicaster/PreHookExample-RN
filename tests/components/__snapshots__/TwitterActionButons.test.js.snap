// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwitterActionButtons component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 30,
        "marginRight": 7,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "uri": "feed_rn_comment_button",
          }
        }
        style={
          Array [
            Object {
              "height": 30,
              "marginRight": 1,
              "width": 30,
            },
            Object {
              "tintColor": "#FFFFFF",
            },
          ]
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
          },
          Object {
            "color": "#FFFFFF99",
          },
        ]
      }
    >
      Reply
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 30,
        "marginRight": 7,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        source={
          Object {
            "uri": "feed_rn_retweet_button",
          }
        }
        style={
          Array [
            Object {
              "height": 30,
              "marginRight": 1,
              "width": 30,
            },
            Object {
              "tintColor": "#FFFFFF",
            },
          ]
        }
      />
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
          },
          Object {
            "color": "#FFFFFF99",
          },
        ]
      }
    />
  </View>
  <EventDetailCount
    count={undefined}
    eventId={undefined}
    label="Favorites"
  />
</View>
`;
